src/main/java/io/codearte/jfairy/producer/company/CompanyProvider.java

```diff
@@ -15,30 +16,68 @@
 	private static final String COMPANY_SUFFIX = "companySuffixes";
 	private static final String COMPANY_EMAIL = "companyEmails";
 
+	private String name;
+	private String domain;
+	private String email;
+	private String vatIdentificationNumber;
+
 	private BaseProducer baseProducer;
 	private DataMaster dataMaster;
 
 	private final VATIdentificationNumberProvider vatIdentificationNumberProvider;
 
 	@Inject
-	public CompanyProvider(BaseProducer baseProducer, DataMaster dataMaster, VATIdentificationNumberProvider vatIdentificationNumberProvider) {
+	public CompanyProvider(BaseProducer baseProducer,
+						   DataMaster dataMaster,
+						   VATIdentificationNumberProvider vatIdentificationNumberProvider,
+						   @Assisted CompanyProperties.CompanyProperty... companyProperties) {
 		this.baseProducer = baseProducer;
 		this.dataMaster = dataMaster;
 		this.vatIdentificationNumberProvider = vatIdentificationNumberProvider;
+
+		for (CompanyProperties.CompanyProperty companyProperty : companyProperties) {
+			companyProperty.apply(this);
+		}
 	}
 
 
 	@Override
 	public Company get() {
 
```

Comment:
imho there are too many conditions in this method and it's too large; maybe you could extract smaller methods and substitute null checks in conditions with elvis operators in return statements where possible?
Please also apply this to other similar places in this PR.